src/pipeline/rackhd/docker/deploy.sh

```diff
@@ -0,0 +1,371 @@
+    cp -f ${on_build_config_dir}/vagrant/config/mongo/config.json ${rackhd_dir}/docker/monorail/config.json
+    #if clone file name is not repo name, this scirpt should be edited.
+    echo "Setup RackHD config done."
+
+}
+
+####################################################
+#
+# Modify the docker-compose.yml default Config file
+#
+####################################################
+setupDockerComposeConfig(){
+
+    local rackhd_docker_images=${1} # the tar file contains all docker images exported by "docker save"
+    local rackhd_dir=${2}
+    local build_record_file=${3}  # the build_record txt generated by previous docker-build steps. if left blank, we will generate it from tar file
+    local build_record=${build_record_file}
+
+    local tmp_file=$(mktemp)
+    # load docker images , for each on-xxx, there will be 2 tags: date-hash and nightly
+    # so in this case, we only put the date-hash tags in the output file( to help to create build_record file)
+
+    echo $SUDO_PASSWORD |sudo -S docker load -i $rackhd_docker_images | grep "UTC" | tee ${tmp_file}
+
+    if [ $? -ne 0 ]; then
+        echo "[Error] Docker Load fail, aborted!"
+        exit -2
+    fi
+    if [ "${build_record_file}" == "" ] ; then
+        local tmp_build_record=$(mktemp)
+        repo_list=""
+        # This path is followed when using the prebuilt images to get image tag
+        while IFS=:  read -r load imagename tag 
+        do
+           repo_list="$repo_list ${imagename}:${tag}"
+        done < ${tmp_file}
+
+        echo $repo_list >> ${tmp_build_record}
+        build_record=${tmp_build_record}
+    fi
+
+    image_list=$(head -n 1 $build_record)
+
+    for repo_tag in $image_list; do
+        repo=${repo_tag%:*}
+        sed -i "s#${repo}.*#${repo_tag}#g" ${rackhd_dir}/docker/docker-compose.yml
+    done
+    echo "Setup docker compose config done."
+}
+
+dockerUp(){
+    local log_dir=$1
+    local rackhd_dir=${2}
+
+    mkdir -p ${log_dir}
+    set +e
+    echo $SUDO_PASSWORD |sudo -S docker pull mongo:latest
+    echo $SUDO_PASSWORD |sudo -S docker pull rabbitmq:management
+    set -e
+
+    echo $SUDO_PASSWORD |sudo -S docker-compose -f ${rackhd_dir}/docker/docker-compose.yml up > ${log_dir}/rackhd.log &
+    echo "Docker up done."
+}
+
+
+##############################################
+#
+# Export docker running log to workspace
+#
+##############################################
+exportLogs(){
+    local log_dir=$1
+    set +e
+    mkdir -p ${log_dir}
+    echo "Export logs start" >> ${log_dir}/export.log
+    containerId=$(echo $SUDO_PASSWORD |sudo -S docker ps|grep "mongo"| head -n1 | awk '{print $1}' )
+    echo "[DEBUG] mongo containerId:${containerId} "
+    #echo $SUDO_PASSWORD |sudo -S docker cp   $containerId:/var/log/mongodb/mongodb.log ${log_dir}
+    if [ -n "$containerId" ]; then
+        echo $SUDO_PASSWORD |sudo -S docker logs $containerId >> ${log_dir}/mongodb.log
```

Comment:
`234     echo $SUDO_PASSWORD |sudo -S docker-compose -f ${rackhd_dir}/docker/docker-compose.yml up >  ${log_dir}/rackhd.log & `

Here is it.